### ISSUE TITLE
Connect-ExchangeOnline: parameter cannot be found that matches parameter name 'ExchangeEnvironmentName'

### ISSUE THREAD

**Issue by JuicyDragon** [MicrosoftDocs/office-docs-powershell#6897]
Hello, I have an automation piece I have written in Java that uses [PowerShellLib](https://github.com/Tuupertunut/PowerShellLibJava) to connect to Exchange Online and Security & Compliance via the [ExchangeOnlineManagement v2.0.3](https://www.powershellgallery.com/packages/ExchangeOnlineManagement/2.0.3) PowerShell Module.

When invoking `Connect-ExchangeOnline` I am receiving the following error in 1 customer's environment (but not my test environment):
`A parameter cannot be found that matches parameter name 'ExchangeEnvironmentName'.`

Originally the command I was issuing was:
`Connect-ExchangeOnline -Credential $credential -EnableErrorReporting -LogDirectoryPath ''D:\O365Logs''`

After receiving the above error, I changed the command to include the parameter mentioned in the error:

`Connect-ExchangeOnline -Credential $credential -EnableErrorReporting -LogDirectoryPath ''D:\O365Logs'' -ExchangeEnvironmentName ''O365Default''`

The customer still receives the error in their environment unfortunately.  The full series of commands is roughly:

```powershell
$PwdSecureString = Get-Content "C:\O365Credentials\username.cred" | ConvertTo-SecureString
$credential = New-Object System.Management.Automation.PSCredential -ArgumentList username, $PwdSecureString
Connect-ExchangeOnline -Credential $credential -EnableErrorReporting -LogDirectoryPath ''D:\O365Logs'' -ExchangeEnvironmentName ''O365Default''
```

Here is a transcript which has a little bit more info regarding the error: [PowershellTranscript.txt](https://github.com/MicrosoftDocs/office-docs-powershell/files/5599656/PowershellTranscript.txt)

Since this is only occurring in 1 out of 3 test environments I have ran this in, my suspicion is this is due to an incomplete configuration on my part.  Unfortunately the error message is not clear to me how I should proceed.  Any suggestions on what I may be doing wrong or whether this may be a bug is much appreciated, thank you!

**Comment by chrisda** [comment]
@JuicyDragon, ExchangeEnvironmentName only works with Connect-ExchangeOnline (ExO PowerShell), not Connect-IPPSSession (SCC PowerShel). You should be using Connect-IPPSSession to connect to SCC PowerShell (even though Connect-ExchangeOnline might also work).

So, I would suggesting using _ConnectionUri_ and _AzureADAuthorizationEndPointUrl_ instead. Both of those parameters are available on Connect-ExchangeOnline and Connect-IPPSSession. And, you don't need to use ANY of these parameters if you're connecting to a 'regular' worldwide tenant. These parameters are only required in organizations like O365 Germany, GCC High, DoD, etc.

**Comment by JuicyDragon** [comment]
Thanks @chrisda.  After I tested a bit more in my test environment and with the customer I learned a few things:
- The error seems to be specifically thrown by line 426 of [ExchangeOnlineManagement.psm1](https://www.powershellgallery.com/packages/ExchangeOnlineManagement/2.0.3/Content/ExchangeOnlineManagement.psm1)
- The error seems to be symptomatic of connectivity errors on this line:
    - Bad credentials
    - Bad proxy configuration
    - Etc...

I had tried with some of the alternate connectivity parameters you mentioned (including using none of them) without success unfortunately.  My customer's environment has pretty strict security so I'm thinking now this error may just be a symptom of something needing to be opened up in their internal security so the call to `New-ExoPSSession` made by `Connect-ExchangeOnline` reaches out to the cloud properly.  It would be handy if this had a more relevant error message, but I understand how these things go.

I think I am good for now, thank you for your assistance! 😄
